src/app/services/yacs.service.ts

```diff
@@ -5,15 +5,22 @@ import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/toPromise';
 
+import { ConflictsService } from './conflicts.service';
+
 @Injectable()
 export class YacsService {
   protected baseUrl = 'http://yacs.cs.rpi.edu/api/v5';
 
-  constructor (private http: Http) {}
+  constructor (private http: Http,
+               private conflictsService: ConflictsService) {}
 
   get (path: string, params: Object = {}): Promise<Object[]> {
     return this.http.get(`${this.baseUrl}/${path}.json?${this.objectToQueryString(params)}`)
                     .toPromise()
+                    .then((response: Response) => {
+                      this.conflictsService.populateConflictsCache(response.json());
```

Comment:
There's a problem here... We don't want to call populateConflictsCache with things that aren't courses. I think this line has to go in the courses controller for now.